File: task1/task1b/Transport.cpp
#include "Transport.h"
#include <iostream>

Transport::Transport(){}

Transport::Transport(int id, const string& type, double speed, double cost): vehicleID(id), vehicleType(type), transportationSpeed(speed), transportationCostPerKm(cost){}


double Transport::calculate_delivery_cost(double distance){
    // Calculate delivery cost based on transportation cost per km and distance
    return transportationCostPerKm * distance;
}

double Transport::calculate_delivery_time(double distance){
    // Calculate estimated delivery time based on transportation speed and distance
    return distance / transportationSpeed;
}


void Transport::display_transport_details(){
    cout << "Vehicle ID: " << getID() << endl;
    cout << "Vehicle Type: " << getType() << endl;
    cout << "Vehicle Transportation Speed: " << getSpeed() << endl;
    cout << "Vehicle Transportation Cost/km: " << getCost() << endl;
}


// Setters
void Transport::setID(int id){
    vehicleID = id;
}

void Transport::setType(const string& type){
    vehicleType = type;
}

void Transport::setSpeed(double speed){
    transportationSpeed = speed;
}

void Transport::setCost(double cost){
    transportationCostPerKm = cost;
}

// Getters
int Transport::getID() const{
    return vehicleID;
}

string Transport::getType() const{
    return vehicleType;
}

double Transport::getSpeed() const{
    return transportationSpeed;
}

double Transport::getCost() const{
    return transportationCostPerKm;
}

File: task2/task2/src/main/java/com/task2/Transport.java
package com.task2;

import java.io.Serializable;

public abstract class Transport implements Serializable{
    protected static final long serialVersionUID = 1L;

    protected int vehicleID;
    protected String vehicleType;
    protected double transportationSpeed;
    protected double transportationCostPerKm;

    public Transport(){

    }

    public Transport(int id, String type, double speed, double cost){
        vehicleID = id;
        vehicleType = type;
        transportationSpeed = speed;
        transportationCostPerKm = cost;
    }

    public abstract String getRoute(int pos);

    public double calculate_delivery_cost(double distance){
        return transportationSpeed * distance;
    }

    public double calculate_delivery_time(double distance){
        return distance/transportationSpeed;
    }

    public void display_vehicle_details(){
        System.out.println("Vehicle ID: " + getID());
        System.out.println("Vehicle Type: " + getType());
        System.out.println("Vehicle Transportation Speed: " + getSpeed());
        System.out.println("Vehicle Transportation Cost/km: " + getCost());
    }

    // Setters
    public void setID(int id){
        vehicleID = id;
    }

    public void setType(String type){
        vehicleType = type;
    }

    public void setSpeed(double speed){
        transportationSpeed = speed;
    }

    public void setCost(double cost){
        transportationCostPerKm = cost;
    }

    // Getters
    public int getID(){
        return vehicleID;
    }

    public String getType(){
        return vehicleType;
    }

    public double getSpeed(){
        return transportationSpeed;
    }

    public double getCost(){
        return transportationCostPerKm;
    }
}
